src/common/webrequest_curl.cpp

```diff
@@ -60,17 +83,74 @@ static size_t wxCURLHeader(char *buffer, size_t size, size_t nitems, void *userd
     return static_cast<wxWebResponseCURL*>(userdata)->CURLOnHeader(buffer, size * nitems);
 }
 
+int wxCURLXferInfo(void* clientp, curl_off_t dltotal,
+                   curl_off_t WXUNUSED(dlnow),
+                   curl_off_t WXUNUSED(ultotal),
+                   curl_off_t WXUNUSED(ulnow))
+{
+    wxCHECK_MSG( clientp, 0, "invalid curl progress callback data" );
+
+    wxWebResponseCURL* response = reinterpret_cast<wxWebResponseCURL*>(clientp);
+    return response->CURLOnProgress(dltotal);
+}
+
+int wxCURLProgress(void* clientp, double dltotal, double dlnow, double ultotal,
+                   double ulnow)
+{
+    return wxCURLXferInfo(clientp, static_cast<curl_off_t>(dltotal),
+                          static_cast<curl_off_t>(dlnow),
+                          static_cast<curl_off_t>(ultotal),
+                          static_cast<curl_off_t>(ulnow));
+}
+
 wxWebResponseCURL::wxWebResponseCURL(wxWebRequestCURL& request) :
     wxWebResponseImpl(request)
 {
+    m_knownDownloadSize = 0;
+
     curl_easy_setopt(GetHandle(), CURLOPT_WRITEDATA, static_cast<void*>(this));
     curl_easy_setopt(GetHandle(), CURLOPT_HEADERDATA, static_cast<void*>(this));
 
+ // Set the progress callback.
+    #if CURL_AT_LEAST_VERSION(7, 32, 0)
+        if ( wxWebSessionCURL::CurlRuntimeAtLeastVersion(7, 32, 0) )
+        {
+            curl_easy_setopt(GetHandle(), CURLOPT_XFERINFOFUNCTION,
+                             wxCURLXferInfo);
+            curl_easy_setopt(GetHandle(), CURLOPT_XFERINFODATA,
+                             static_cast<void*>(this));
+        }
+        else
+        {
+            curl_easy_setopt(GetHandle(), CURLOPT_PROGRESSFUNCTION,
+                             wxCURLProgress);
+            curl_easy_setopt(GetHandle(), CURLOPT_PROGRESSDATA,
+                             static_cast<void*>(this));
+        }
+    #else
+        curl_easy_setopt(GetHandle(), CURLOPT_PROGRESSFUNCTION, wxCURLProgress);
+        curl_easy_setopt(GetHandle(), CURLOPT_PROGRESSDATA,
+                         static_cast<void*>(this));
+    #endif
```

Comment:
```suggestion
        else
    #endif
        {
            curl_easy_setopt(GetHandle(), CURLOPT_PROGRESSFUNCTION,
                             wxCURLProgress);
            curl_easy_setopt(GetHandle(), CURLOPT_PROGRESSDATA,
                             static_cast<void*>(this));
        }
```
This allows to avoid duplicating the last 2 calls.

Reply:
done in 3e5fd54